src/thunderbots/software/backend_output/grsim/grsim_backend.cpp

```diff
@@ -58,12 +63,46 @@ void GrSimBackend::sendGrSimPacket(const grSim_Packet& packet)
         remote_endpoint, 0, err);
 }
 
+
 void GrSimBackend::sendPrimitives(
     const std::vector<std::unique_ptr<Primitive>>& primitives)
 {
+    std::vector<grSim_Packet> grsim_packets;
+
+    MotionController::Velocity robot_velocities;
+    grSim_Packet grsim_packet;
+
+
+    // get timestamp before running bang-bang controller for all robots
+    auto bangbang_timestamp = std::chrono::steady_clock::now();
+
+    std::chrono::duration<double> delta_time;
+
     // TODO: Implement this
-    // https://github.com/UBC-Thunderbots/Software/issues/21
-    grSim_Packet grsim_packet = createGrSimPacket(
-        0, YELLOW, Point(0.5, -0.1), Angle::ofRadians(-0.8), 4.0, true, false);
+    // https://github.com/UBC-Thunderbots/Software/issues/99
+    for (auto& prim : primitives)
+    {
+        MovePrimitive movePrim = dynamic_cast<MovePrimitive&>(*prim);
+
+        auto current_time =
+            std::chrono::steady_clock::now();  // get the current time right before
+                                               // running bang-bang to get a time-delta
+                                               // for acceleration
+
+        delta_time = current_time - bangbang_timestamp;
```

Comment:
Won't this `delta_time` be a _tiny_ value, because we're only taking the difference in time between a few lines of code? I would expect this to be the `delta_time` between "now" and when the robot was last updated / last ran the motion controller.

Reply:
Yeah, the first time it is run delta_time is just the time for a few lines of code. The only solution I can think of to prevent this is having a global struct/passing a struct that contains a timestamp for each robot of when the motion controller was last run.

Reply:
I think a "global" timestamp variable is best, since there's no benefit to having a separate timestamp for each robot (since they'd all effectively be the same time since they're in the same loop in the same function), and managing multiple timestamps is tricky when robots are added or removed.

I'd recommend a `static` variable in the `sendPrimitives` function. That way it can maintain state between function calls but isn't "global" to the entire file/program.

Reply:
Added static timestamp to sendPrimitves.